From 4722727373533b53489b66d3436b50ac156f23bf Mon Sep 17 00:00:00 2001
From: Jackie Dong <xy-jackie@139.com>
Date: Mon, 14 Jul 2025 17:46:55 +0800
Subject: [PATCH] ALSA: hda/realtek: Support mute LED for Yoga with ALC287
Git-commit: 4722727373533b53489b66d3436b50ac156f23bf
Patch-mainline: v6.16-rc7
References: jsc#PED-14430 jsc#PED-14297 jsc#PED-14024

Support mute LED on keyboard for Lenovo Yoga series products with
Realtek ALC287 chipset.

Tested on Lenovo Slim Pro 7 14APH8.

[ slight comment cleanup by tiwai ]

Signed-off-by: Jackie Dong <xy-jackie@139.com>
Link: https://patch.msgid.link/20250714094655.4657-1-xy-jackie@139.com
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/pci/hda/patch_realtek.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 132cef8fab08..47ceb6f32291 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -7497,6 +7497,9 @@ static void alc287_fixup_yoga9_14iap7_bass_spk_pin(struct hda_codec *codec,
 	};
 	struct alc_spec *spec = codec->spec;
 
+	/* Support Audio mute LED and Mic mute LED on keyboard */
+	hda_fixup_ideapad_acpi(codec, fix, action);
+
 	switch (action) {
 	case HDA_FIXUP_ACT_PRE_PROBE:
 		snd_hda_apply_pincfgs(codec, pincfgs);
-- 
2.52.0

